Add support for modular import of language files

KelvenCassamo opened the pull request:
This pull request implements modular language file import functionality, allowing translation files to be split into separate files for each language. Starting with version 1.0.3, translations can be loaded in a modular way, which makes it easier to maintain large translation files.

Additionally, the `getWord` function has been enhanced to support multiple tenses for verb translations. The function now handles various verb forms such as infinitive, past, participle, and gerund. This provides more flexibility for handling dynamic verb conjugation within translations.

The documentation has been updated to reflect these new features, including usage examples of the modular import functionality and the `getWord` feature.

**Main Changes**:
- Support for modular import of language files via the `<import-language>` tag  [See here](https://github.com/KelvenCassamo/java-language-system/tree/feature-verb-translation?tab=readme-ov-file#211-importing-individual-language-files-available-from-version-103).
- Updated the `getWord` function to support verb tenses: infinitive, past, participle, and gerund.
- Updated the README.md file with new examples and documentation improvements.
- Example of how to use individual language files and the `getWord` feature for verb translations [See here](https://github.com/KelvenCassamo/java-language-system/tree/feature-verb-translation?tab=readme-ov-file#28-word-translation-with-specific-forms).

**To Test**:
1. Download version 1.0.3 or higher.
2. Use the `languages.xml` example to import translations from separate files.
3. Test the `getWord` functionality to check verb translations in different tenses using the newly implemented features.
4. Verify that translations load correctly from modular files and that verb forms are returned as expected.

KelvenCassamo committed:
Add support for importing individual language files

Previously, all languages were defined in a single file, making it difficult to manage and scale. This update introduces the ability to reference separate XML files for each language using <import-language> tags.

Example:
<languages>
  <import-language file='languages/english.xml'/>
  <import-language file='languages/portuguese.xml'/>
  <import-language file='languages/spanish.xml'/>
</languages>

Key changes:
- Updated XML parser to handle <import-language> tags.
- Added support for loading and merging language data from multiple files.
- Improved modularity and maintainability of language definitions.

KelvenCassamo committed:
Add support for Modular Language Import from version 1.0.3

The Modular Language import feature is now available, allowing for modular XML imports starting from version 1.0.3. See documentation for more details.

KelvenCassamo committed:
Updated README with essential badges and improved content

bivdev commented on src/mz/cassamo/jls/LanguageHandler.java:
```diff
@@ -66,9 +71,23 @@ class LanguageHandler extends DefaultHandler {
+                try {
+                    initFromFile(importFilePath);
+                } catch (IOException e) {
+                    System.err.println("Error while importing file: " + importFilePath);
```
In the following block, the error message for a failed file import is printed directly using System.err.println:
````java
System.err.println("Error while importing file: " + importFilePath);
````
To ensure that these messages are only logged during debugging, you could wrap this System.err.println statement in a condition that checks whether debug mode is enabled. For example:
```java
  if(LanguageSystem.isDebugMode()){
      System.err.println("Error while importing file: " + importFilePath);
    }
```

bivdev reviewed (changes requested):
Great work on the implementation of the modular language file import and the enhancements to the `getWord` function! The code structure is clean, and the logic for processing language imports is straightforward.

#### Key Suggestions:

- As noted in the inline comment, consider wrapping the `System.err.println` statement in a conditional check for debug mode. This will ensure that error messages are only logged during debugging, keeping the production logs clean and manageable.
- If not already present, it may be beneficial to centralize logging functionality (e.g., using a logging framework like `java.util.logging` or `SLF4J`) to handle debug and error messages more efficiently.

KelvenCassamo committed:
Wrap error logging in debug mode check for file import failures

- Added a conditional check using `LanguageSystem.isDebugMode()` to ensure that error messages for file import failures are only logged during debugging.
- Improves log cleanliness and prevents unnecessary error logs in production.

KelvenCassamo reviewed (commented):
I edited the logging logic to wrap the System.err.println statement with a check for LanguageSystem.isDebugMode()
